addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -1879,44 +1879,65 @@ registry.TextHighlight = publicWidget.Widget.extend({
      * @override
      */
     async start() {
+        // We need to adapt the text highlights on resize (E.g. custom fonts
+        // loading, layout option changes, window resized...), mainly to take in
+        // consideration the rendered line breaks in text nodes... But after
+        // every adjustment, the `ResizeObserver` will unfortunately immediately
+        // notify a size change once new highlight items are observed leading to
+        // an infinite loop. To avoid that, we use a lock map (`observerLock`)
+        // to block the callback on this first notification for observed items.
+        this.observerLock = new Map();
+        this.resizeObserver = new window.ResizeObserver(entries => {
+            window.requestAnimationFrame(() => {
+                const topTextEls = new Set();
+                entries.forEach(entry => {
+                    const target = entry.target;
+                    const topTextEl = target.closest(".o_text_highlight") || target.querySelector(".o_text_highlight");
+                    if (topTextEl) {
+                        if (this.observerLock.get(target)) {
+                            // Unlock the top highlight element along with its
+                            // items (lines), the next resize will trigger a
+                            // highlight adaptation.
+                            return this.observerLock.set(target, false);
+                        }
+                        topTextEls.add(topTextEl);
+                        if (!target.classList.contains("o_text_highlight")) {
+                            // Unobserve the target (it will be replaced by
+                            // a new one after the update).
+                            this.resizeObserver.unobserve(target);
+                        }
+                    }
+                });
```

Comment:
I don't understand this code.
- You have a `target` which is the element who was being observed and thus which got resized
- You search for the closest text_hightlight element ancestors find then inside the target itself
- If there is one, you unlock the target... not the text_highlight element you found? While the `_lockHighlightObserver` seems to lock only element which are text_highlight themselves
- Then if the target is not the text_highlight element itself, you stop observing it... which case is that? A element which was a text_highlight, got obserbed but somehow now contains or is in another text hightlight but is not a text highlight itself? :thinking:

Reply:
@qsm-odoo Here is a summary of the `resizeObserver`'s logic (after adding some updates & fixes):

**[1]- First step (on public widget `start()`):**

A- The highlights are applied > `applyTextHighlight()`;

B- For every `.o_text_highlight`, its parent **[x]** and its `.o_text_highlight_item` elements are locked > `_lockHighlightObserver()`

C- For every `.o_text_highlight`, its parent **[x]** and its `.o_text_highlight_item` elements are observed > `_observeHighlightResize()`

D- Right after the `resizeObserver.observe()`, [the resize handler will be called automatically even if there is no "resize"](https://drafts.csswg.org/resize-observer/#intro).

E- This first handler call should just unlock "observed targets":
```javascript
if (this.observerLock.get(target)) {
    return this.observerLock.set(target, false);
}
```
**[x]:** Fixed: The `resizeObserver` will simply ignore an element if it has an inline display. We should actually target the closest non-inline parent (see: `_closestToObserve()`).

**[2]- On every "RESIZE":**

A- Resize handler should get the `.o_text_highlight` items (inside the target if it's a parent or the closest one if the target is a highlight item...) and add them to the adaptation `Set()`.

B- We unobserve all highlight lines before the update since they will be automatically removed from the DOM (because of highlight lines computing...).

C- After `switchTextHighlight()`, we will automatically go to [1.B].

To respond to your questions:

> You have a `target` which is the element who was being observed and thus which got resized
> You search for the closest text_hightlight element ancestors find then inside the target itself
- If the target is a parent, we need to get the targeted ".o_text_highlight" elements **[xx]**, otherwise, it's the `closest()` one (see: [1.C]).

**[xx]**: Fixed: If the target is a parent, we should get all highlight elements, not just the first one.

> If there is one, you unlock the target... not the text_highlight element you found? While the _lockHighlightObserver seems to lock only element which are text_highlight themselves
- `_lockHighlightObserver()` locks all observed targets, (see: [1.D-E]) 

> Then if the target is not the text_highlight element itself, you stop observing it... which case is that? A element which was a text_highlight,
got obserbed but somehow now contains or is in another text hightlight but is not a text highlight itself? 🤔
- Good catch! Actually, the condition was from an old code version where only the content inside `.o_text_highlight` was observed and it was used to target the `.o_text_highlight_item` lines (see: [2.B]). Fixed now :+1:

**Remark:** I added a separate commit to allow the translation of text options (text animation & highlights) in stable as a fix for **opw-3686777**. Just let me know if something needs to be adjusted here :+1:

Reply:
@xO-Tx Thanks :+1: I quickly flew over the code, I'll not check all details again, seems good :+1: 

> Remark: I added a separate commit to allow the translation of text options (text animation & highlights) in stable as a fix for opw-3686777. Just let me know if something needs to be adjusted here 👍

Could you move that to its own PR? I'd like to comment some things on that one. Also, should not it be for 16.0 in the end? But maybe we can indeed just hide the text animation button in 16.0 and take the risk in 17.0 only.

Reply:
> Could you move that to its own PR? I'd like to comment some things on that one. Also, should not it be for 16.0 in the end? But maybe we can indeed just hide the text animation button in 16.0 and take the risk in 17.0 only.

The code will be added in the `17.0` forward port of: https://github.com/odoo/odoo/pull/157087. It targets `15.0` now (where the "Text Animations" were available) and, as suggested, the option button will be disabled for `15.0~16.0`.
I also fixed another issue that makes animated text invisible in "translate" mode in the same PR. If something needs to be changed, just let me know :+1: